server/channels/app/notification_push.go

```diff
@@ -768,6 +772,37 @@ func (a *App) SendTestPushNotification(deviceID string) string {
 	return "true"
 }
 
+func (a *App) SendTestMessage(c request.CTX, userID string) (*model.Post, *model.AppError) {
```

Comment:
Since this is all centered around creating a custom post that forces a notification, I feel like this belongs in the `post.go` file or in the API itself, it's not notification specific.